### PR TITLE
parse extra keyword args in class definition

### PR DESCRIPTION
**Pull request by pkienzle**
mypy cannot parse the classes defined in typing.py, such as:

    class Tuple(Final, metaclass=TupleMeta, _root=True): ...

Instead, it stops with a parse error at the comma following metaclass=TupleMeta.  The original code assumed that there would be a list of superclasses, possibly followed by metaclass=....  This patch generalizes the parsing to an arbitrary set of keyword arguments, popping the metaclass out of the list when it is complete.

I updated the ClassDef node in nodes.py to hold on to the keyword arguments and serialize to and from json, but I don't know how to trigger the serialization code, so this is not tested.

### CODE REVIEW

File: mypy/nodes.py [hunk 1]
```diff
@@ -670,6 +674,7 @@ def deserialize(self, data: JsonDict) -> 'ClassDef':
                        Block([]),
                        [mypy.types.TypeVarDef.deserialize(v) for v in data['type_vars']],
                        metaclass=data['metaclass'],
+                       kwargs=dict((k, v.deserialize()) for k, v in data['kwargs']),
```
Comment (gvanrossum) [inline mypy/nodes.py]: That's not how you invoke deserialize() -- it is a class method.

**Comment by gvanrossum** [comment]
As you can tell from running mypy on itself (which is what the Travis CI test does) there are a few problems with this code, but nothing that can't be fixed!

I also wonder if it's worth fixing this right now, since @ddfisher is working on a complete rewrite of the parser (see #1353). The new parser will be based on Python's ast module so it will automatically support this. (Well, mypy still needs to type-check the keyword args and possibly understand how they affect a class definition, but that's secondary -- first it needs to not reject the syntax.)

However if you want to learn something about mypy I encourage you to persist and finish this PR!

**Comment by pkienzle** [comment]
I simplified the patch so that it is equivalent to the fastparse handling of class definitions.  That is, the ClassDef node includes the metaclass but not any other keyword args and this patch doesn't change that.

**Comment by gvanrossum** [comment]
LGTM. Can you please add a test showing that it ignores other keywords?

**Comment by pkienzle** [comment]
Done.

**Comment by gvanrossum** [comment]
Thanks!

**Pull request merged**